x/schema/keeper/msg_server_schema.go

```diff
@@ -69,6 +68,39 @@ func (k msgServer) CreateSchema(goCtx context.Context, msg *types.MsgCreateSchem
 }
 
 func (k msgServer) DeprecateSchema(goCtx context.Context, msg *types.MsgDeprecateSchema) (*types.MsgDeprecateSchemaResponse, error) {
-	// TODO: implement
-	return nil, errors.New("unimplemented")
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	err := msg.ValidateBasic()
+	if err != nil {
+		return nil, err
+	}
+
+	schemas, foundSchemas := k.GetWhatIsFromCreator(ctx, msg.GetCreator())
```

Comment:
@joshLong145 Have you made a `GetWhatIsByDID` method?

Reply:
@mcjcloud its fine, parsing and querying by did on chain is more unnecessary compute/gas, keep it this way and we can utilize did for network level resolution